client/src/components/Countdown.js

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect } from 'react'
+import { useState, useContext } from 'react';
+import { CallerContext } from '../containers/Home';
+// import { removeTrack } from '../services/CallService'
+
+function Countdown(props) {
+
+  const { callHasExpired } = useContext(CallerContext)
+
+  const timeSubString = props.timeData.callLength.substring(3)
+  const startingTime = Date.parse('1970-01-01T00:' + timeSubString + 'Z')
+  const interval = 1000;
+
+  const [time, setTime] = useState(startingTime)
+
+  useEffect(() => {
+
+    setInterval(() => {
```

Comment:
Using an interval for a countdown makes it depend on the processing speed of the computer (and on how busy it is). It can happen that when busy, the time passes slower...

Instead set an expire time and in the interval (or a set of setTimeouts) you check the time left.

Reply:
Excellent point - will refactor.